### DISCUSSION TITLE
Upgrade to 2.1.2 not working

### DISCUSSION THREAD

**Discussion by StefanGlaesser** [modoboa/modoboa#2993] [Help] [answered]
Hello,

I've installed a fresh server with Ubuntu 22.04 and Modoboa 2.1.1 with the installer (https://github.com/modoboa/modoboa-installer)

Now I tried to upgrade to 2.1.2 but it's not doing like I expected. I'm still stuck on 2.1.1. That's what I'm doing:

    sudo -u modoboa -i bash
    source virtenv/bin/activate
    pip install modoboa==2.1.2
    python manage.py migrate

That fails due missing modules. So just installing them:

    pip install modoboa_amavis
    pip install modoboa_contacts
    pip install modoboa_radicale
    pip install modoboa_postfix_autoreply
    pip install modoboa_sievefilters
    pip install modoboa_webmail
    pip install psycopg2

Then the migrate-script seems to work:

    python manage.py migrate
    Operations to perform:
       Apply all migrations: admin, auth, authtoken, contenttypes, core, dmarc, dnstools, imap_migration, lib, limits, maillog, modoboa_amavis, modoboa_contacts, modoboa_postfix_autoreply, modoboa_radicale, otp_static, otp_totp, relaydomains, reversion, sessions, sites, transport
    Running migrations:
      No migrations to apply.

Then running 

     python manage.py collectstatic

     You have requested to collect static files at the destination location as specified in your settings:

     /srv/modoboa/instance/sitestatic

     This will overwrite existing files!
     Are you sure you want to do this?
     
     Type 'yes' to continue, or 'no' to cancel: no

Answering 'yes' overwrite some files in the sitestatic folder.

Last step:

     python manage.py check --deploy

Results into:   

```
System check identified some issues:
WARNINGS:
?: (modoboa-amavis.W001) AMAVIS_DEFAULT_DATABASE_ENCODING does not match the character encoding used by the Amavis database.
	HINT: Check your database character encoding and set/update AMAVIS_DEFAULT_DATABASE_ENCODING.
?: (security.W001) You do not have 'django.middleware.security.SecurityMiddleware' in your MIDDLEWARE so the SECURE_HSTS_SECONDS, SECURE_CONTENT_TYPE_NOSNIFF, SECURE_BROWSER_XSS_FILTER, SECURE_REFERRER_POLICY, and SECURE_SSL_REDIRECT settings will have no effect.
?: (security.W009) Your SECRET_KEY has less than 50 characters, less than 5 unique characters, or it's prefixed with 'django-insecure-' indicating that it was generated automatically by Django. Please generate a long and random SECRET_KEY, otherwise many of Django's security-critical features will be vulnerable to attack.

System check identified 3 issues (3 silenced).
```
As described here (https://modoboa.readthedocs.io/en/latest/upgrade.html) now just restarting the webserver but it's still on version 2.1.1. What's going wrong? Any hints for me? 

Best regards,
Stefan

**Comment by Spitfireap** [comment]
Hi, can you restart `uwsgi` service and check the reported version again ?

**Reply by StefanGlaesser** [reply]
I did that already. Also restarted the whole machine. Still on 2.1.1. :-(

**Reply by Spitfireap** [reply, marked as answer]
`source virtenv/bin/activate` are you sure this command worked ? installer will create the environment under `/srv/modoboa/env` by default (so `env/bin/activate`)

**Reply by StefanGlaesser** [reply]
That was the problem. As it was not documented, I've created my own virtenv. So would be nice, that it will be documented that "env" is the default virtual enviroment. :-)
